pkg/istio-agent/agent.go

```diff
@@ -371,24 +371,25 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 	// Are we listening to the Istio default SDS server socket, or something else?
 	isIstioSDS := configuredAgentSocketPath == security.GetIstioSDSServerSocketPath()
 
+	disableCaClient := false
+
 	socketExists, err := checkSocket(ctx, configuredAgentSocketPath)
 	if err != nil {
 		return nil, fmt.Errorf("failed to check SDS socket: %v", err)
 	}
 	if socketExists {
-		log.Infof("Existing workload SDS socket found at %s. Default Istio SDS Server won't be started", configuredAgentSocketPath)
-	} else {
+		log.Infof("Existing workload SDS socket found at %s. Default Istio SDS Server will only serve files", configuredAgentSocketPath)
+		disableCaClient = true
```

Comment:
(WIP so ignore if already intend to change) probably can just set `a.secOpts.ServeOnlyFiles` directly here.